src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/TerminateDebugHandler.ts

```diff
@@ -0,0 +1,43 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as psList from 'ps-list';
+import { DebugSession } from 'vscode';
+
+import { RazorLogger } from '../RazorLogger';
+
+export async function onDidTerminateDebugSession(
+  event: DebugSession,
+  logger: RazorLogger,
+  targetPid: number | undefined,
+) {
+  if (event.type !== 'pwa-chrome' && event.type !== 'pwa-msedge') {
+    return;
+  }
+  if (!targetPid) {
+    return;
+  }
+
+  logger.logVerbose(`[DEBUGGER] Terminating debugging session with PID ${targetPid}...`);
+
+  let processes: psList.ProcessDescriptor[] = [];
+  try {
+    processes = await psList();
+  } catch (error) {
+    logger.logError(`Error retrieving processes under PID ${targetPid} to clean-up: `, error);
+  }
+
+  try {
+    process.kill(targetPid);
+    processes.map((proc) => {
```

Comment:
>processes.map((proc) => { [](start = 4, length = 25)

Rather than using `map` I think you want to instead loop through `processes` to find `targetPid`. Then loop through just the remaining entries in the map to find any child processes. This way you will avoid the process id reuse problem with ppid.

Reply:
I'll add this in a future PR.